controllers/watcher_controller.go

```diff
@@ -246,7 +253,7 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 
 	subLevelSecretName, err := r.createSubLevelSecret(ctx, helper, instance, transporturlSecret, inputSecret, db)
 	if err != nil {
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: time.Duration(60) * time.Second}, nil
 	}
 	// the subLevelSecretName will be the value for the Secret field in the
```

Comment:
I just realized that the `subLevelSecretName` variable is unused, in `ensureAPI` we simply use the fact the secret is created with the same name of the instance. I think either way is fine, but in a follow-up we should change the function to stop returning it if we're not going to get used, or use is when creating the subCRs instances